### ISSUE TITLE
Could not add rust-analysis by full name

### ISSUE THREAD

**Issue by sophiajt** [rust-lang/rustup#1077]
`rustup component list` shows a full name of components (like **rust-analysis-i686-pc-windows-msvc**, **rust-std-aarch64-apple-ios** and so on).

But `rustup component add rust-analysis-i686-pc-windows-msvc` itself does not work:

> error: toolchain 'nightly-i686-pc-windows-msvc' does not contain component 'rust-analysis-i686-pc-windows-msvc' for target 'i686-pc-windows-msvc'

However it works with the short name: `rustup component add rust-analysis`.

Full log:

```
> rustup show
...
active toolchain
----------------

nightly-i686-pc-windows-msvc (default)
rustc 1.18.0-nightly (7627e3d31 2017-04-16)

> rustup component list
cargo-i686-pc-windows-msvc (default)
rls-i686-pc-windows-msvc
rust-analysis-i686-pc-windows-msvc
rust-docs-i686-pc-windows-msvc (default)
rust-src
rust-std-aarch64-apple-ios
...

> rustup component add rust-analysis-i686-pc-windows-msvc
error: toolchain 'nightly-i686-pc-windows-msvc' does not contain component 'rust-analysis-i686-pc-windows-msvc' for target 'i686-pc-windows-msvc'
```

(from: https://github.com/rust-lang-nursery/rls/issues/262)

**Comment by Diggsey** [comment]
Yeah, this is poor UX, it affects docs too, and possibly libstds. It should be possible to install alternate targets by passing `--target xyz` to `rustup component add rust-analysis`

**Comment by brson** [comment]
Yeah since this affects several commands and lists, needs a bit of a think before deciding what to do.

**Comment by pravic** [comment]
> error: toolchain 'stable-i686-pc-windows-msvc' does not contain component 'rust-analysis-i686-pc-windows-msvc' for target 'i686-pc-windows-msvc'

Since the target is already known, why not just split the component name? Or if it is combined (full), do not make full name for such commands and leave it as `rust-analysis`, `rust-src` and so on.

**Comment by kinnison** [comment]
I believe this was resolved by #2088
